src/adapter/src/catalog.rs

```diff
@@ -5548,7 +5542,14 @@ impl Catalog {
         pcx: Option<&PlanContext>,
     ) -> Result<CatalogItem, AdapterError> {
         let mut session_catalog = self.for_system_session();
-        enable_features_required_for_catalog_open(&mut session_catalog);
+        let mut state = session_catalog.state.clone();
```

Comment:
Can you add a comment to this with the GH issue link for hierarchical variables, indicating that we can remove the clones once the issue is done.